### DISCUSSION TITLE
select vs projection

### DISCUSSION THREAD

**Discussion by brightchip** [Automattic/mongoose#10291] [Q&A] [answered]
Please, someone tell me the differences
Between  [projection](https://mongoosejs.com/docs/api/query.html#query_Query-projection)

And  [select](https://mongoosejs.com/docs/api.html#query_Query-select)

Which one return only the filtered fields and not filtering in memory

**Comment by vkarpov15** [comment, marked as answer]
Two names for the same thing. The only difference is that `query.projection(v)` overwrites the current projection with `v`, whereas `query.select(v)` adds `v` to the current projection like [`Object.assign()`](https://masteringjs.io/tutorials/fundamentals/assign).

**Reply by eduboardio** [reply]
Thanks
